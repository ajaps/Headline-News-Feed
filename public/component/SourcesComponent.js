import React from 'react';

import * as myActions from '../actions/HeadlineActions';

export default class SourcesComponent extends React.Component {

  setSources(sources) {
    myActions.getArticles(sources);
  }

  render() {
    return (
      <li><a onClick={this.setSources.bind(this, this.props.id)}>{this.props.name}</a></li>
    );
  }
}
