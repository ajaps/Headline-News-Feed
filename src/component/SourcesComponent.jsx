import React from 'react';
import classNames from 'classnames';

import * as myActions from '../actions/HeadlineActions';
import ArticleStore from '../stores/Article';
/**
 * Represents a Source.
 */
export default class SourcesComponent extends React.Component {

/**
 * calls an action to set sources url and available sorting filter for articles
 * @param {String} sources The selected source id
 * @param {String} sortAvailable The selected sort Filter
 * @return {void}
 */
  setSources(sources, sortAvailable) {
    const url = ArticleStore.getArticleUrl;
    myActions.getArticles(url, sources, sortAvailable);
  }

/**
 * @returns {component} A component with relevant source.
 */
  render() {
    // sets class name based on user source selection
    const btnClass = classNames({
      sourceBorder: this.props.sourceHigh[0] === this.props.id,
    });

    return (
      <li className={btnClass} >
        <a onClick={this.setSources.bind(this, this.props.id, this.props.sortBysAvailable)}>
          {this.props.name}
        </a>
      </li>
    );
  }
}
