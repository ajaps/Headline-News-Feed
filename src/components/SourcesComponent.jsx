import classNames from 'classnames';
import PropTypes from 'prop-types';
import React from 'react';
import ReactLoading from 'react-loading';
import * as myActions from '../actions/HeadlineActions';
import ArticleStore from '../stores/Articles';
/**
 * Represents a Source.
 */
export default class SourcesComponent extends React.Component {
  constructor(props) {
    super(props);
    this.setSources = this.setSources.bind(this);
    this.searchSource = this.searchSource.bind(this);
    this.state = { searchTerm: '' };
  }

/**
   * sets the an object in the state to the value of the search string
   * @param {Object} e The calling object property
   * @return {void}
   */
  searchSource(e) {
    const searchString = e.target.value;
    this.setState({
      searchTerm: searchString,
    });
  }

/**
 * calls an action to set sources url and available sorting filter for articles
 * @param {Object} e The calling object property
 * @return {void}
 */
  setSources(e) {
    const sort = e.target.dataset.sort.split(',');
    const url = ArticleStore.getArticleUrl;
    myActions.fetchArticles(url, e.target.dataset.id, sort);
  }

/**
 * @returns {component} A component with relevant source.
 */
  render() {
    const error = this.props.error;
    console.log(error);
    const sources = this.props.sources;
     // Search through sources, return all if search term is empty('')
    const sourcesComponents = sources.map((sourcesItem) => {
      // sets class name based on user source selection using classnames module
      const btnClass = classNames({
        sourceBorder: this.props.sourceSelected === sourcesItem.id,
      });
      const reg = RegExp(this.state.searchTerm, 'gi');
      if (sourcesItem.name.search(reg) !== -1) {
        return (
          <li className={btnClass} key={sourcesItem.id}>
            <a onClick={this.setSources} data-id={sourcesItem.id}
            data-sort={sourcesItem.sortBysAvailable}> {sourcesItem.name} </a>
          </li>
        );
      }
      return null;
    });

    return (
      error ?
      <div className="col-sm-3 col-md-2 sidebar well">
        <h4 className="sourcesHeader"> News Sources </h4>
        <input type="text" className="form-control"
        placeholder="Search Sources..." onChange={this.searchSource} /><br/>
        <ul className="nav nav-sidebar Source-Container">
          <ReactLoading className="spin" type="spin" color="#3b5998"/>
        </ul>
      </div> :
      <div className="col-sm-3 col-md-2 sidebar well">
        <h4 className="sourcesHeader"> News Sources </h4>
        <input type="text" className="form-control"
        placeholder="Search Sources..." onChange={this.searchSource} /><br/>
        <ul className="nav nav-sidebar Source-Container">
          {sourcesComponents}
        </ul>
      </div>
    );
  }
}

SourcesComponent.propTypes = {
  sourceSelected: PropTypes.string.isRequired,
  error: PropTypes.string,
  sources: PropTypes.array.isRequired,
};

