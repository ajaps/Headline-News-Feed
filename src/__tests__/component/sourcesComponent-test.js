import { mount, shallow } from 'enzyme';
import React from 'react';
import toJson from 'enzyme-to-json';
import SourcesComponent from '../../component/SourcesComponent';

jest.mock('../../apiFolder/NewsApi', () => ({
  getData: () => Promise.resolve('getPromise')
}));

describe('Category', () => {
  let app;
  const sourceObject =
    {
      "id": "abc-news-au",
      "name": "ABC News (AU)",
      "description": "Australia's most trusted source of local, national and world news. Comprehensive, independent, in-depth analysis, the latest business, sport, weather and more.",
      "url": "http://www.abc.net.au/news",
      "category": "general",
      "language": "en",
      "country": "au",
      "urlsToLogos": {
      "small": "",
      "medium": "",
      "large": ""
      },
        "sortBysAvailable": [
        "top"
      ]
    };

  it('should render as expected', () => {
    app = shallow(<SourcesComponent key={sourceObject.id}{...sourceObject}/>);
    const tree = toJson(app);
    expect(tree).toMatchSnapshot();
  });

  it('should have extacly one anchor element', () => {
    expect(app.find('a').length).toEqual(1);
  });

  it('should have display text using the value of "name" in the object passed', () => {
    const result = 'ABC News (AU)';
    expect(app.text()).toBe(result);
  });

  it('should call the setCategory function on click the anchor element', () => {
    app.find('a').simulate('click');
    expect(app).toBeTruthy();
  });
});
