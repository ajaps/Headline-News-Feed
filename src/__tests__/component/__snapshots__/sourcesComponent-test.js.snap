// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Category should render as expected 1`] = `
<li>
  <a
    onClick={[Function]}
  >
    ABC News (AU)
  </a>
</li>
`;
